pkg/keyspace/gc.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keyspace
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/kvproto/pkg/keyspacepb"
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/member"
+	"github.com/tikv/pd/pkg/storage/kv"
+	"go.uber.org/zap"
+)
+
+// Some common per-loop config.
+const (
+	// gcPerLoopTimeout specifies timeout for a single loop, it must be less than minKeyspaceGCInterval (10m).
+	gcPerLoopTimeout = 5 * time.Minute
+	// gcKeyspaceBatch specifies batch size when fetching keyspaces.
+	gcKeyspaceBatch = 100
+)
+
+// gcWorker is used to clean up keyspace related information.
+type gcWorker struct {
+	sync.RWMutex
+	manager *Manager
+	member  *member.EmbeddedEtcdMember
+
+	nextKeyspaceID uint32
+	ticker         *time.Ticker
+	// gcLifeTime specifies how long should we keep archived keyspace.
+	gcLifeTime time.Duration
+}
+
+// newGCWorker returns a newGCWorker.
```

Comment:
Introduce the concept of 'archiving' instead of calling it GC to prevent other developers from confusing this GC with MVCC GC

Reply:
Additionally, do not use the concept of 'GC' when archiving a keyspace, and the naming of safepoint variables should include a prefix to indicate what operation the safepoint corresponds to.

Reply:
ok, sounds good, will use `cleaning` & `archiving` instead of `gc`